### PR TITLE
Feat: Order suite

### PR DESCRIPTION
**Pull request by essteer**
Adds the Order class and related repo, service and controller layers.

Adds unit test classes for entity and service layers. All tests passing.

Additional mappings are saved in draft form on the branch `feat-orders-extended-mappings`.

### CODE REVIEW

File: src/main/java/com/fdmgroup/forex/enums/OrderStatus.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.fdmgroup.forex.enums;
+
+public enum OrderStatus {
+    ACTIVE, CANCELLED, CLEARED, CLOSED, EXPIRED
```
Comment (dexterklui) [inline src/main/java/com/fdmgroup/forex/enums/OrderStatus.java:12]: Do you think it's a good idea to add JavaDoc here to explain what each enum mean?
Especially CANCELLED vs CLEARED vs CLOSED

Comment (essteer) [reply src/main/java/com/fdmgroup/forex/enums/OrderStatus.java:12]: Yes, will do

Comment (essteer) [reply src/main/java/com/fdmgroup/forex/enums/OrderStatus.java:12]: See [593c591](https://github.com/forex-agile/backend/pull/8/commits/593c59162fe1595128ade621afe5f4076d010789)

File: src/main/java/com/fdmgroup/forex/enums/OrderStatus.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.fdmgroup.forex.enums;
+
+/**
+ * Denotes the state of the order as follows:
+ * ACTIVE: Order is live and open to matches
+ * CANCELLED: Order cancelled by the user
```
Comment (dexterklui) [inline src/main/java/com/fdmgroup/forex/enums/OrderStatus.java]: Order can be CANCELLED on InsufficientFundException too, when no partial trades ever happened

Comment (essteer) [reply src/main/java/com/fdmgroup/forex/enums/OrderStatus.java]: [a687131](https://github.com/forex-agile/backend/pull/8/commits/a687131dfe403beabca4640ce17d97442725c208)

File: src/main/java/com/fdmgroup/forex/DataLoader.java [hunk 3]
```diff
@@ -33,4 +63,30 @@ private void createRoles() {
 			roleRepo.save(new Role(defaultRole));
 	}
 
+	@Transactional
+	private void createSampleData() {
+		Role role = roleRepo.findByRole(defaultRole).get();
+
+		Currency hkd = currencyRepo.findById(usdCode).orElseGet(() -> {
```
Comment (essteer) [inline src/main/java/com/fdmgroup/forex/DataLoader.java]: This should be `hkd = currencyRepo.findById(hkdCode)` -> currently reads `usdCode`

**Review by essteer — COMMENTED** [review]
@dexterklui This looks fine to me - just the one fix as per the comment. I can't mark this as "Request changes" for some reason

**Review by dexterklui — APPROVED** [review]

**Pull request merged**